utils/vlog/tests/fig41-full-adder/CMakeLists.txt

```diff
@@ -0,0 +1,10 @@
+add_file_target(FILE adder.sim.v SCANNER_TYPE verilog)
+v2x(NAME adder TOP_MODULE adder SRCS adder.sim.v)
+
+add_file_target(FILE golden.pb_type.xml SCANNER_TYPE xml)
+add_file_target(FILE golden.model.xml SCANNER_TYPE xml)
+
+xml_sort(NAME pb_type_golden_sort FILE golden.pb_type.xml OUTPUT adder.pb_type.golden.xml)
+xml_sort(NAME pb_type_actual_sort FILE adder.pb_type.xml  OUTPUT adder.pb_type.actual.xml)
+
+diff(NAME adder_diff GOLDEN adder.pb_type.golden.xml ACTUAL adder.pb_type.actual.xml)
```

Comment:
This seems very heavy, do I just define a function which is something like
```cmake
# A test for the Verilog to XML conversion. It compares the output to the golden files.
v2x_test(adder.sim.v)
```

Reply:
sure can!

Reply:
I recommend the v2x_test look like:

v2x_test(
  NAME <name>
  V2X_SRC <src>
  GOLDEN_PB_TYPE_XML <file>
  GOLDEN_MODEL_XML <file>
  )